src/sourceMachine.ts

```diff
@@ -123,6 +135,45 @@ export const makeSourceMachine = (auth: SupabaseAuthClient) => {
         },
       },
       states: {
+        checking_if_on_legacy_url: {
```

Comment:
Wouldnt it be much less involved if we'd just use Next's redirects for that? 🤔 

A config like this could work:
```js
  redirects() {
    return [
      {
        source: '/viz',
        has: [
          {
            type: 'query',
            key: 'id',
            value: '(?<id>.*)',
          },
        ],
        destination: '/viz/:id',
        basePath: false,
        permanent: true,
      },
    ];
  },
```

(I couldn't make it work without `basePath: false`)

This leaves the `id` param in the query though - but that can be just cleaned up client-side, outside of the machine (during the initialization time of the module).

It just seems like such a minor thing and outside of the business logic that externalizing this from the machine seems like a win for me.

Reply:
Great idea!

Reply:
@Andarist This results in a URL which looks like this:

`http://localhost:3000/viz/a3613bbd-aeff-4618-a7e7-08d11395ec79?id=a3613bbd-aeff-4618-a7e7-08d11395ec79`

Which I suppose is fine

Reply:
@mattpocock 

> This leaves the id param in the query though - but that can be just cleaned up client-side, outside of the machine (during the initialization time of the module).

Reply:
I don't care about this being removed or not though, like you say - it rly doesn't matter